src/test/java/de/tum/in/www1/artemis/exam/ExamIntegrationTest.java

```diff
@@ -405,26 +470,76 @@ void testUpdateExam_asInstructor() throws Exception {
         // Note: ZonedDateTime has problems with comparison due to time zone differences for values saved in the database and values not saved in the database
         assertThat(exam).usingRecursiveComparison().ignoringFields("id", "course", "endDate", "startDate", "visibleDate").isEqualTo(createdExam);
         assertThat(examCountBefore + 1).isEqualTo(examRepository.count());
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void testUpdateExam_failsWithoutCourse() throws Exception {
         // No course is set -> bad request
-        exam = ExamFactory.generateExam(course1);
+        Exam exam = ExamFactory.generateExam(course1);
         exam.setId(1L);
         exam.setCourse(null);
         request.put("/api/courses/" + course1.getId() + "/exams", exam, HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void testUpdateExam_failsWithExamIdMismatch() throws Exception {
         // Course id in the updated exam and in the REST resource url do not match -> bad request
-        exam = ExamFactory.generateExam(course1);
+        Exam exam = ExamFactory.generateExam(course1);
         exam.setId(1L);
         request.put("/api/courses/" + course2.getId() + "/exams", exam, HttpStatus.BAD_REQUEST);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideExamsWithInvalidDates")
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
```

Comment:
Please move this test towards line 430 since it also used the same method source.

Reply:
Hm really want to keep creates / updates together

Reply:
We should rather think about splitting this whole thing up as it's really huge. One good candidate would be a separate suspicious behaviour integration test class